pgmpy/utils/state_name.py

```diff
@@ -69,7 +69,12 @@ def get_state_no(self, var, state_name):
         Given `var` and `state_name` return the state number.
         """
         if self.state_names:
-            return self.name_to_no[var][state_name]
+            try:
+                return self.name_to_no[var][state_name]
+            except KeyError:
+                raise KeyError(
+                    f"{state_name} is an unknown state name. It must be one of {list(self.name_to_no[var].keys())}"
```

Comment:
I think it might be nicer if we also mention the variable for which the state_name is not known. Something like: `state: {state_name} is unknown for variable: {var}. ...... `. Might be helpful in debugging when multiple variables were passed to the original method.

Reply:
Thanks! Fixed in commit 3b24eb72c401811bdd600c9b41eec26ad21491e1